homeassistant/components/device_tracker/icloud.py

```diff
@@ -78,6 +79,14 @@ def update_icloud(now):
         except PyiCloudNoDevicesException:
             _LOGGER.info('No iCloud Devices found!')
 
+    if name is None:
```

Comment:
`name` will never be `None` if you follow my other comment so you can simplify this again.